### PR TITLE
Issue#932

### PR DESCRIPTION
**Pull request by nagu165**
Problem :

When editing a file system, the form shows a disabled selector for the mount point field. Using a disabled widget does not make sense in this case because the mount point will never be editable. The disabled widget should be replaced by plain text.

Moreover, the selector currently uses a FormSelect component. It should be replaced by the Select component, similar to the file system type selector.

Solution :

- Updated the MountPointFormSelector to use Select component Instead of the previous FormSelect component.

- The select component adds more customizability and some advanced features as well.

- Updated the mountpoint field to show the mountpath when currentVolume is defined and prompts the user to choose the mountpoint from the list of volume templates in case the currentVolume is undefined.

Testing :

Tested manually

Screenshots:

Before:
Screenshot 2024-01-09 232254
Screenshot 2024-01-09 232710

After:
image

image

### CODE REVIEW

**Comment by nagu165** [comment]
I am really sorry about the previous pull request, I accidentally force-pushed it. Sorry for the trouble.

Thank you for your understanding.

File: web/package/cockpit-agama.changes [hunk 1]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Fri Jan 19 09:34:26 UTC 2024 -  Nagu <nagukalakuntla@gmail.com>
+
+- Storage UI: show mount point selector only when the user can change it.
+  (gh#openSUSE/agama#973)
```
Comment (dgdavid) [inline web/package/cockpit-agama.changes:5]:
```suggestion
  (gh#openSUSE/agama#1007)
```

**Review by dgdavid — APPROVED** [review]
@nagu165 

In its current shape, the PR (former #973)  LGTM :+1:, thanks a ton for your interest in the project and, especially, for your contribution.

It has taken more time than both expected, though :sweat_smile: :sweat_smile: However, beyond the contribution to the Agama project, hopefully it has been useful to your career. As advice for the future, don't forget to take as much time as necessary for understanding both, the tools you're using and the task you are trying to accomplish. No matter if you're contributing to a project or doing something for yourself. Taking care of the details is just as important as having a good grasp of the tool you are using at the time.

Too many words from my end already :stuck_out_tongue_winking_eye: 

 Let's merge it!

**Comment by nagu165** [comment]
> @nagu165
> 
> In its current shape, the PR (former #973) LGTM 👍, thanks a ton for your interest in the project and, especially, for your contribution.
> 
> It has taken more time than both expected, though 😅 😅 However, beyond the contribution to the Agama project, hopefully it has been useful to your career. As advice for the future, don't forget to take as much time as necessary for understanding both, the tools you're using and the task you are trying to accomplish. No matter if you're contributing to a project or doing something for yourself. Taking care of the details is just as important as having a good grasp of the tool you are using at the time.
> 
> Too many words from my end already 😜
> 
> Let's merge it!

Thank you so much for your guidance sir, As you have mentioned I learned a lot during this project and will strive to keep learning more.

**Comment by ancorgs** [comment]
JFYI, this pull request was mentioned in the [announcement of Agama 8](https://yast.opensuse.org/blog/2024-05-17/agama-8).

Thanks for your contribution!

**Pull request merged**